src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -3921,6 +3939,24 @@ void Unit::_UpdateAutoRepeatSpell()
     }
 }
 
+bool Unit::CanSparringWith(const Unit* attacker) const
+{
+    if(GetTypeId() != TYPEID_UNIT || IsCharmedOwnedByPlayerOrPlayer())
+        return false;
+
+    if(!attacker)
+        return false;
+
+    if(attacker->GetTypeId() != TYPEID_UNIT || attacker->IsCharmedOwnedByPlayerOrPlayer())
+        return false;
+
+    if(Creature const* me = ToCreature())
+        if(me->GetSparringPct() == 0.0f)
+            return false;
```

Comment:
spacing with if.

!IsUnit()

!attacker->IsUnit()

Reply:
First we doesn't have IsUnit() then here it is: IsCreature (that we don't have also) because a "Unit" can return: TYPEID_UNIT or TYPEID_PLAYER.